src/analysis/retail/warrior/arms/modules/core/Execute/MortalStrike.tsx

```diff
@@ -0,0 +1,148 @@
+import { t } from '@lingui/macro';
+import { formatPercentage } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/warrior';
+import { SpellLink } from 'interface';
+import Analyzer, { SELECTED_PLAYER, Options } from 'parser/core/Analyzer';
+import { calculateMaxCasts } from 'parser/core/EventCalculateLib';
+import Events, { CastEvent } from 'parser/core/Events';
+import Abilities from 'parser/core/modules/Abilities';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+
+import ExecuteRange from './ExecuteRange';
+
+class MortalStrikeAnalyzer extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+    executeRange: ExecuteRange,
+  };
+
+  protected abilities!: Abilities;
+  protected executeRange!: ExecuteRange;
+
+  private mortalStrikeCasts = 0;
+  private mortalStrikesOutsideExecuteRange = 0;
+  private mortalStrikesInExecuteRange = 0;
+
+  get mortalStrikeUsageThresholds() {
+    const cd = this.abilities.getAbility(SPELLS.MORTAL_STRIKE.id)?.cooldown || 6;
+    const max = calculateMaxCasts(
+      cd,
+      this.owner.fightDuration, //- this.executeRange.executionPhaseDuration(),
+      //--NOTE: you no longer stop using mortal strike during execute
+    );
+
+    return {
+      actual: this.mortalStrikeCasts / max,
+      isLessThan: {
+        minor: 0.9,
+        average: 0.8,
+        major: 0.7,
+      },
+      style: ThresholdStyle.PERCENTAGE,
+    };
+  }
+
+  //this bit is irrelevant now.
+  // You want to keep Deep wounds active on your target when in execution phase, without overcasting Mortal Strike
+  //get notEnoughMortalStrikeThresholds() {
+  //  const cd = 12000; //Deep wounds duration
+  //  const max = calculateMaxCasts(cd, this.executeRange.executionPhaseDuration());
+  //  const maxCast =
+  //    this.mortalStrikesInExecuteRange / max > 1 ? this.mortalStrikesInExecuteRange : max;
+
+  //  return {
+  //    actual: this.mortalStrikesInExecuteRange / maxCast,
+  //    isLessThan: {
+  //      minor: 0.9,
+  //      average: 0.8,
+  //      major: 0.6,
+  //    },
+  //    style: ThresholdStyle.PERCENTAGE,
+  //  };
+  //}
+
+  //get tooMuchMortalStrikeThresholds() {
+  //  const cd = 12000; //Deep wounds duration
+  //  const max = calculateMaxCasts(cd, this.executeRange.executionPhaseDuration());
+  //  const maxCast =
+  //    this.mortalStrikesInExecuteRange / max > 1 ? this.mortalStrikesInExecuteRange : max;
+
+  //  return {
+  //    actual: 1 - this.mortalStrikesInExecuteRange / maxCast,
+  //    isGreaterThan: {
+  //      minor: 1,
+  //      average: 1.15,
+  //      major: (maxCast + 1) / maxCast,
+  //    },
+  //    style: ThresholdStyle.PERCENTAGE,
+  //  };
+  //}
```

Comment:
if these are irrelevant, you can delete. we can pull old stuff out of git history as needed.